core/pfe-core/controllers/timestamp-controller.ts

```diff
@@ -2,6 +2,10 @@ import type { ReactiveController, ReactiveElement } from 'lit';
 
 export type DateTimeFormat = 'full' | 'long' | 'medium' | 'short';
 
+const options = ['dateFormat', 'timeFormat', 'customFormat', 'displaySuffix', 'locale', 'relative', 'utc', 'hour12'] as const;
```

Comment:
```suggestion
export interface Options {
  dateFormat: DateTimeFormat;
  // etc
}

const defaults = {
  dateFormat: 'short',
  timeFormat: 'short',
} // etc
```

let the constructor take a `Partial<Options>` in it's second parameter, and have setting one of those request an update. here's one way you could do that:

```js
constructor(host, options) {
  this.#options = {}
  for (const [name, value] of Object.entries(defaults)) {
    this.#options[name] = options[name] ?? value;
    Object.defineProperty(this, 'name', {
      get() { return this.#options[name]; },
      set(v) { 
        this.#options[name] = v;
        host.requestUpdate();
      },
    });
  }
```